lib/Index/Index.cpp

```diff
@@ -962,7 +966,18 @@ class IndexSwiftASTWalker : public SourceEntityWalker {
   bool startEntityDecl(ValueDecl *D);
 
   bool reportRelatedRef(ValueDecl *D, SourceLoc Loc, bool isImplicit, SymbolRoleSet Relations, Decl *Related);
-  bool reportRelatedTypeRef(const TypeLoc &Ty, SymbolRoleSet Relations, Decl *Related);
+
+  /// Report references for dependent types
+  ///
+  /// NOTE: If the dependent type is a typealias, report the underlying types as well.
+  ///
+  /// \param Ty The type being referenced.
+  /// \param Relations The relationship between the referenced type and the passed Decl.
+  /// \param Related The Decl that is referencing the type.
+  /// \param isImplicit Whether the reference is implicit, such as for a typealias' underlying type.
+  /// \param Loc The location of the reference, otherwise the location of the TypeLoc is used.
+  bool reportRelatedTypeRef(const TypeLoc &Ty, SymbolRoleSet Relations, Decl *Related,
+                            bool isImplicit=false, Optional<SourceLoc> Loc=Optional<SourceLoc>());
```

Comment:
```suggestion
                            bool isImplicit=false, Optional<SourceLoc> Loc = {});
```
If you realllllllllly want to keep the Optional.

Reply:
ah sorry i lost the thread on that in the const convo, removed